src/Controller/UserController.php

```diff
@@ -290,6 +290,40 @@ public function block(User $blockee, Request $request, EntityManager $em) {
         ]);
     }
 
+    /**
+     * @IsGranted("ROLE_USER")
+     * @Entity("blockee", expr="repository.findOneOrRedirectToCanonical(username, 'username')")
+     *
+     * @param User          $blockee
+     * @param Request       $request
+     * @param EntityManager $em
+     *
+     * @return Response
+     */
+    public function blockJSON(User $blockee, Request $request, EntityManager $em) {
```

Comment:
why is this one has JSON in the name, but others don't?
I think we can stick to normal naming and then just use @return Response - JSONResponse for AJAX calls.
In the comments above the method to show that this is AJAX method.

Reply:
This method had JSON in the name because this method is used in the ajax request for blocking a user in the report window. There is a separate block method that is used as a separate page, so I couldn't just call it `block` because of a naming conflict.  What should I do for this?

Reply:
JSON works. 

Adding few TODOs around both methods with “name these better, this one creates the block, bla bla, this one does that, balls”

Reply:
*bla bla
On the phone sorry -can’t edit comment